spec.html

```diff
@@ -879,6 +879,12 @@ <h1>Algorithm Conventions</h1>
     <p>Algorithm steps may declare named aliases for any value using the form “Let _x_ be _someValue_”. These aliases are reference-like in that both _x_ and _someValue_ refer to the same underlying data and modifications to either are visible to both. Algorithm steps that want to avoid this reference-like behaviour should explicitly make a copy of the right-hand side: “Let _x_ be a copy of _someValue_” creates a shallow copy of _someValue_.</p>
     <p>Once declared, an alias may be referenced in any subsequent steps and must not be referenced from steps prior to the alias's declaration. Aliases may be modified using the form “Set _x_ to _someOtherValue_”.</p>
 
+    <emu-clause id="sec-parallelism">
+      <h1>Parallelism</h1>
+      <p>A step that ends with “<dfn>in parallel</dfn>” specifies that it is run at the same time as subsequent steps in the algorithm. This specification does not define the precise mechanism by which this is achieved. For example, an implementation may use processes or threads that make use of multiple CPUs or multiple cores of a single CPU. If a parallel step is subdivided, the substeps run sequentially within the unit of parallelism of the parallel step.</p>
+      <p>A parallel step exhibits a data race if it modifies an alias or mutates a specification or ECMAScript language value at the same time as any other step. Parallel steps generally require synchronization, such as critical sections, to avoid data races. This specification does not define the precise mechanism by which synchronization is achieved. For example, an implementation may use locks that guarantee mutual exclusion. It is an editorial error for a parallel step to exhibit a data race.</p>
```

Comment:
So the editors are saying that, in this spec, no parallel step exhibits a data race (and if they're wrong, the spec is broken in some sense). But the paragraph also talks about how an implementation can avoid data races: why would an implementation need to avoid something that you assert doesn't happen?

Also, you might want to point out that this concept of data race is similar to but distinct from the one defined in "Memory Model".

Reply:
I'm saying the spec uses synchronization in parallel steps to avoid data races. The waiter stuff currently already has a critical section. If there is a data race, that is a spec bug. The implementation bit is just saying you can implement what the spec just refers to as a "critical section" with a mutex.

Reply:
Removed parallel steps entirely.